nntrainer/layers/layer_node.cpp

```diff
@@ -287,6 +306,15 @@ bool LayerNode::getFlatten() const {
   return flatten.get();
 }
 
+unsigned int LayerNode::getNumOutputTensors() const {
+  auto &output = std::get<props::OutputSpec>(*layer_node_props);
```

Comment:
can this mismatch with getNumOutputs()?

Reply:
I intentionally give this different name. Cause getNumOutputs is to get the connection number and this if get the number of output tensors. eventually total output size must be getNumOutputs x getNumOutputTensors.

Reply:
I think there is some confusion. `getNumOutputs` gives the number of outputs from the init context, which is the number of output dimensions. The number of connections is given by `getNumOutputConnections`.
So maybe getNumOutputs and getNumOutputsTensors should be in sync.

Reply:
AFIK, both of them are the same for the current implementation. even if it is called differently. Cause the number of output of init_context is set by the number of the connection actually. What I'm thinking is we have to return the number of how many layers are connected if the getNumOutputConnections is called, and if you call the getNumOutputs() then you have to give the number of connection x number of output tensors. and if you call the getNumOutputTesnors(), it has to return the number of output tensors for each output.

Reply:
I think I am missing something. In my understanding, in a graph I thought we only had 1 tensor for each connection?
For if there are two nodes N1 and N2, both feeding their outputs to N3, then both N1 and N2 can only give 1 tensor to N3, making total of 2 tensors received by N3. Please correct me if I am wrong.